### ISSUE TITLE
Improve the posts queries

### ISSUE THREAD

**Issue by bragaz** [desmos-labs/desmos#966]
## Feature description 
Currently it's not possible to query posts by user. We should give this possibility to the devs. Especially those working with smart-contracts bindings will benefit from it.

## Implementation proposal
I was thinking that something similar to what it's done inside the [reactions query](https://github.com/desmos-labs/desmos/blob/b5738646ff3ff03f864ab423fdc556b319d7d45b/x/reactions/keeper/grpc_query.go#L42) is feasible also here.

What do you guys think? @RiccardoM @dadamu .

**Comment by dadamu** [comment]
Since there would be millions of posts inside a subspace and no post-user pair index inside db now. The query for a user will be super heavy, costing O(n) time complexity, like cosmos-sdk [`delegations`](https://github.com/cosmos/cosmos-sdk/blob/main/x/staking/keeper/grpc_query.go#L103) query which blocks bdjuno before.

I recommend the user register with post id to a smart contract for checking the user post, instead.

**Comment by RiccardoM** [comment]
I agree with @dadamu here, this is not easy to implement and it would make nodes crash very easily. Unless a benchmark can prove otherwise, I would not implement this feature.

**Comment by bragaz** [comment]
Ok I see. I also guess this is something doable for clients through GQL right? @RiccardoM
